app/modules/nep9/Nep9QrGenerator.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+/* eslint-disable */
+import QRCode from 'qrcode/lib/browser'
+import nep9 from './nep9'
+import tokenList from '../../core/tokenList.json'
+import { imageMap } from '../../assets/nep5/raw-svg-export'
+
+const TOKENS = Object.keys(tokenList)
+  .map(key => tokenList[key])
+  .reduce((accum, token) => {
+    accum[token.networks[1].hash] = token
+    return accum
+  }, {})
+
+export default class Nep9QrGenerator {
+  static parseUri(uri: string) {
+    return nep9.parseUri(uri)
+  }
+
+  // $FlowFixMe
+  constructor({ nep9Data, width = 250, canvasEl, imgEl }) {
+    let canvas
+    // $FlowFixMe
+    this.uri = nep9.generateUri(nep9Data)
+    const options = {
+      errorCorrectionLevel: 'H',
+      width
+    }
+
+    // $FlowFixMe
+    this.creationPromise = Promise.all([
+      Promise.resolve(TOKENS),
+      new Promise((resolve, reject) => {
+        QRCode.toCanvas(
+          // $FlowFixMe
+          this.uri,
+          options,
+          (err, canvas) => (err ? reject(err) : resolve(canvas))
+        )
+      })
+    ])
+      .then(([nep5Data, c]) => {
+        canvas = c
+        const token = {}
+        if (nep9Data.asset === 'NEO') token.symbol = 'NEO'
+        if (nep9Data.asset === 'GAS') token.symbol = 'GAS'
+        else {
+          const foundToken =
+            Object.keys(tokenList)
+              .map(key => tokenList[key])
+              .find(token => token.networks[1].hash === nep9Data.asset) || {}
+          token.symbol = foundToken.symbol
+        }
+        const logo = imageMap[token.symbol] || imageMap.NEO
```

Comment:
same comment as above

Reply:
I think we are ok here without `get` because this token will always default to an empty object meaning that the lookup against `imageMap`  would be `imageMap[undefined]` which will not throw an exception